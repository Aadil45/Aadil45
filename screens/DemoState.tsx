import React,{Component} from 'react';
import { View, Text} from 'react-native';
class DemoState extends Component {
  constructor(){
    super();
    this.state={
      data:'test',
    }
  }
    Change(x){
      // this.setState({data:x*5})
      return x*10
    }
    abc(){
      alert('Hello World');
    }

    render() {
      return (
         <View>
            <Text>Welcome</Text>
         </View>
      );
    }
}
export default DemoState