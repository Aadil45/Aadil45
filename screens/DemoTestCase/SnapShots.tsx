import React from 'react';
import {StyleSheet, View,Image } from 'react-native';
const SnapShots: React.FC = () => {
  
  
  return (
      <View 
    style={styles.main}
    >
      <View style={{ flex: 1, backgroundColor: "#FF9933",borderTopLeftRadius:50,borderTopRightRadius:50 }} />
      <View style={{ flex: 1, backgroundColor: "#fff",justifyContent:'center',alignItems:'center' }} >
        <Image source={require('../../assets/Images/AshokChakra.png')} style={{width:'50%',height:'100%'}} />
      </View>
      <View style={{ flex: 1, backgroundColor: "#138808",borderBottomLeftRadius:50,borderBottomRightRadius:50 }} />
    {/* <Profile data='heloo'/> */}
     </View>
  );
};
export default SnapShots;
const styles = StyleSheet.create({
  main: {
    flex: 1,
    flexDirection: 'column',
    padding: 30,
  },
  box: {
    flex: 1,
    backgroundColor: 'red',
  },
  input: {
    borderColor: 'black',
    marginTop:10,
    borderWidth: 0.5,
    marginHorizontal: 5,
    paddingHorizontal: 20,
    height: 40,
    width: '100%',
  },
});
