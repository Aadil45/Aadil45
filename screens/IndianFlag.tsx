import React, {useState} from 'react';
import {StyleSheet, View, Text,Image, TextInput} from 'react-native';
// import { TextInput } from 'react-native-gesture-handler';
/* function change(x){
  return x*10;
 }
 module.exports = change; */
const IndianFlag: React.FC = props => {
  const [userName,setUserName]=useState('')
  
  function abc(){
    alert('Hello world ')
  }
  return (
      <View 
    style={styles.main}
    >
{/*       <View style={{ flex: 1, backgroundColor: "#FF9933",borderTopLeftRadius:50,borderTopRightRadius:50 }} />
      <View style={{ flex: 1, backgroundColor: "#fff",justifyContent:'center',alignItems:'center' }} >
        <Image source={require('../assets/Images/AshokChakra.png')} style={{width:'50%',height:'100%'}} />
      </View>
      <View style={{ flex: 1, backgroundColor: "#138808",borderBottomLeftRadius:50,borderBottomRightRadius:50 }} /> */}
     {/* <Text>{props.data}</Text> */}
      <TextInput style={styles.input} testID={'username'} value={userName} onChangeText={setUserName}  />
    </View>
  );
};
export default IndianFlag;
const styles = StyleSheet.create({
  main: {
    flex: 1,
    flexDirection: 'column',
    padding: 30,
  },
  box: {
    flex: 1,
    backgroundColor: 'red',
  },
  input: {
    borderColor: 'black',
    marginTop:10,
    borderWidth: 0.5,
    marginHorizontal: 5,
    paddingHorizontal: 20,
    height: 40,
    width: '100%',
  },
});
