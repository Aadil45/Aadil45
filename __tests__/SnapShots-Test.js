import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

import SnapShots from '../screens/DemoTestCase/SnapShots';
// import Home from '../screens/Home';
import DemoState from '../screens/DemoState';
// import change from '../screens/IndianFlag';
import IndianFlag from '../screens/IndianFlag';
describe('Main testing', () => {
  test('Snapshot testing', () => {
    const snap = renderer.create(<SnapShots />).toJSON();
    //  const snap= renderer.create(<Home />).toJSON();
    expect(snap).toMatchSnapshot();
  });
  
  /* it('function and state testing', () => {
    const HomeData = renderer.create(<DemoState />).getInstance();
    console.log(HomeData);
    // expect(HomeData.change(2)).toEqual(10);
    expect(HomeData.abc.length).toBe(0);
    //  HomeData.change(2)
    //   expect(HomeData.state.data).toEqual(10) 
  }); */
  /*  it('function testing ',()=>{
    expect(change(2)).toEqual(20)
  }); */
  /* let findElement = function (tree, element) {
    for (node in tree.children) {
      console.log('tree.children[node]', tree.children[node]);
      if (tree.children[node].props.testID == element) return true;
    }
    return undefined;
  };
  test('find value', () => {
    const HomeData = renderer.create(<IndianFlag />).toJSON();
    expect(findElement(HomeData, 'username')).toBeDefined();
  }); */
  /*   it('props',()=>{
    const wrapper=shallow(<IndianFlag data='hello'/>).props();
    // const Data=shallow(<IndianFlag data='hello'/>)
    console.log('data',wrapper.children[0].props.children);
    expect(wrapper.children[0].props.children).toEqual('hello')
  }) */
});
