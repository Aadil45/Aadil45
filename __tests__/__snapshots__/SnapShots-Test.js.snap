// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main testing Snapshot testing 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "padding": 30,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FF9933",
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 50,
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/Images/AshokChakra.png",
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "50%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#138808",
        "borderBottomLeftRadius": 50,
        "borderBottomRightRadius": 50,
        "flex": 1,
      }
    }
  />
</View>
`;
